py_zipkin/zipkin.py

```diff
@@ -290,26 +291,28 @@ def stop(self, _exc_type=None, _exc_value=None, _exc_traceback=None):
         popped off. The actual logging of spans depends on sampling and that
         the logging was correctly set up.
         """
-        # Always remove the stored zipkin_attrs
         if self.do_pop_attrs:
             pop_zipkin_attrs()
 
-        # Exit early if this request is not being sampled
         if not self.zipkin_attrs or not self.zipkin_attrs.is_sampled:
             return
 
-        # If this is the root span, exit the context (which will handle logging)
+        # Logging context is only initialized for "root" spans of the local
+        # process (i.e. this zipkin_span not inside of any other local
+        # zipkin_spans)
         if self.logging_context:
             self.logging_context.stop()
             self.logging_context = None
             return
 
+        # If we've gotten here, that means that this span is a child span of
+        # this context's root span (i.e. it's a zipkin_span inside another
+        # zipkin_span).
         end_timestamp = time.time()
 
-        # Put the old parent_span_id back on the handler
         self.log_handler.parent_span_id = self.old_parent_span_id
 
-        # To get a full span we just set cs=sr and ss=cr.
+        # We are simulating a full two-part span locally, so set cs=sr and ss=cr
```

Comment:
In previous iterations we were just logging ('cs', 'cr') for these spans (which is why we call them "client" spans in a lot of the code). I can't recall the exact issue, but we found this didn't play nicely with the UI, so we started logging ('sr', 'ss') as well. So to me logging both client and server annotations felt a bit like cheating - the operation the span was representing is entirely local but the span contained both server and client annotations.